### ISSUE TITLE
SSL_ERROR_NO_CYPHER_OVERLAP on some Brazilian government websites

### ISSUE THREAD

**Issue by Refold3547** [mullvad/mullvad-browser#327]
Some important websites developed by the Brazilian Government are displaying the error described in the title. This is the case of https://lattes.cnpq.br

![lattes cnpq](https://github.com/user-attachments/assets/c70c7c30-efab-44ba-8e46-83b3abb415f8)

This issue specifically occurs in Mullvad Browser as the site works fine in other browsers. I am using the latest version (v.13.5.3)

**Comment by Thorin-Oakenpants** [comment]
this is likely  because we lock "non-modern" [1] cipher suites to false
- https://gitlab.torproject.org/tpo/applications/tor-browser/-/blob/tor-browser-128.2.0esr-14.0-1/browser/app/profile/001-base-profile.js#L131-140

[1] their usage is very low (but not low enough for Firefox to deprecate/disable), see Moz Telemetry, or they lack PFS (perfect forward secrecy)

fyi: @PieroV

**Comment by Thorin-Oakenpants** [comment]
^ security.ssl3.rsa_aes_128_sha ?

**Comment by PieroV** [comment]
I think they're using sha1 :fearful:.

**Comment by Thorin-Oakenpants** [comment]
https://gitlab.torproject.org/tpo/applications/tor-browser/-/blob/tor-browser-128.2.0esr-14.0-1/browser/app/profile/001-base-profile.js#L131-140

these are all default false in FF nightly
```
pref("security.ssl3.dhe_rsa_aes_128_sha", false, locked);
pref("security.ssl3.dhe_rsa_aes_256_sha", false, locked);
pref("security.ssl3.ecdhe_ecdsa_aes_256_sha", false, locked);
pref("security.ssl3.ecdhe_ecdsa_aes_128_sha", false, locked);
```

the resr
```
pref("security.ssl3.ecdhe_rsa_aes_128_sha", false, locked); <-- this
pref("security.ssl3.ecdhe_rsa_aes_256_sha", false, locked); <-- this
pref("security.ssl3.rsa_aes_128_sha", false, locked);
pref("security.ssl3.rsa_aes_256_sha", false, locked);
```

setting `security.ssl3.rsa_aes_128_sha` + `security.ssl3.rsa_aes_256_sha` to false in FF reproduces the error for this particular website

**Comment by Thorin-Oakenpants** [comment]
so a year ago - see [here](https://gitlab.torproject.org/tpo/applications/tor-browser/-/issues/40183#note_2849563) - moz telemetry said these two ciphers were (the first number is the telemetry code)
-  61 : 0.25% : TLS_RSA_WITH_AES_128_CBC_SHA
-  63 : 0.27% : TLS_RSA_WITH_AES_256_CBC_SHA

That would have dropped a little. And of course it's along thin tail - i.e the more popular the site, the more likely it isn't affected - e.g. alexa top 1mn ... that said, sucks when it's an essential site such as bank ...  but this is not going to be changed. The payoff is too minimal to justify weakening security, given our threat model

**Comment by Thorin-Oakenpants** [comment]
> I think they're using sha1

yes, that's what the `_sha` in the prefs mean

**Comment by hackademix** [comment]
It seems we've been a bit too strict with disabling cyphersuites. Some of them (including those needed for this site) can be re-enabled for now, see discussion at https://gitlab.torproject.org/tpo/applications/mullvad-browser/-/issues/361#note_3089049

**Comment by ruihildt** [comment]
Fixed in 13.5.7
